src/modalities/utils/verify_tokenization_constistency.py

```diff
@@ -0,0 +1,198 @@
+import json
+import os
+import pickle
+import tempfile
+import warnings
+from enum import Enum
+from pathlib import Path
+from typing import Callable
+
+import sentencepiece as spm
+import tqdm
+from transformers import AutoTokenizer
+
+from modalities.api import create_raw_data_index, pack_encoded_data
+from modalities.dataloader.dataset import PackedMemMapDatasetBase
+
+
+class TokenizerTypes(Enum):
+    sentence_piece = "sentence_piece"
+    hugging_face = "hugging_face"
+
+
+def _run_tokenization(
+    src_path: Path, index_path: Path, pbin_path: Path, eod_token: str, tokenizer_config: dict, jq_pattern: str = ".text"
+):
+    # create index
+    create_raw_data_index(src_path=src_path, index_path=index_path)
+    # run tokenization
+    num_cpus = os.cpu_count()
+
+    tokenization_config_dict = {
+        "settings": {
+            "src_path": src_path,
+            "dst_path": pbin_path,
+            "index_path": index_path,
+            "jq_pattern": jq_pattern,
+            "num_cpus": num_cpus,
+            "eod_token": eod_token,
+            "processing_batch_size": 10,
+            "raw_samples_queue_size": 300,
+            "processed_samples_queue_size": 300,
+        },
+        "tokenizer": {**tokenizer_config},
+    }
+
+    pack_encoded_data(config_dict=tokenization_config_dict)
+
+
+def _verify_index(src_path: Path, index_path: Path):
+    with open(src_path, "rb") as f:
+        jsonl_binary_string = f.read()
+
+    with open(src_path, "rb") as f:
+        binary_string_list = f.readlines()
+
+    with open(src_path, "r", encoding="utf-8") as f:
+        string_list = f.readlines()
+
+    with open(index_path, "rb") as f:
+        jsonl_index = pickle.load(f)
+
+    for i, (offset, length) in tqdm.tqdm(enumerate(jsonl_index), desc="Verifying index"):
+        # check that the index works correctly on the binary data
+        binary_string = binary_string_list[i]
+        if binary_string_list[i].endswith(b"\n"):
```

Comment:
```suggestion
        if binary_string.endswith(b"\n"):
```
Just for improved readability and simplicity